src/sidebar/store/modules/drafts.js

```diff
@@ -117,10 +119,14 @@ function createDraft(annotation, changes) {
  * An empty draft has no text and no reference tags.
  */
 function deleteNewAndEmptyDrafts() {
+  /**
+   * @param {import('redux').Dispatch} dispatch
+   * @param {() => { drafts: State }} getState
+   */
   return (dispatch, getState) => {
     const newDrafts = getState().drafts.filter(draft => {
       return (
-        metadata.isNew(draft.annotation) &&
+        !draft.annotation.id &&
```

Comment:
Can you tell me a little bit about why you elected not to use the `isSaved` abstraction here? Obviously all that function encapsulates is the presence of a service-assigned `id`, but it helps express context.

Reply:
`isSaved` currently expects a full annotation, whereas `draft.annotation` only has the ID fields.